### PR TITLE
ci: improve sshnpd health check

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
**- What I did**

- sshnpd now outputs all (stdout and stderr) into `/atsign/all.txt`
- also prints to stdout

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by xavierchanth** [comment]
@JeremyTubongbanua try the posix compliant notation: `<command> > all.txt 2>&1`

File: tests/end2end_tests/tests/service-container-sshnpd.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     networks:
       - sshnpd
     healthcheck:
-      test: ["CMD", "grep", "-Eq", "monitor started for @", "/atsign/err.txt"]
```
Comment (xavierchanth) [inline tests/end2end_tests/tests/service-container-sshnpd.yaml:8]: What's the reason for not suppressing output?

Comment (JeremyTubongbanua) [reply tests/end2end_tests/tests/service-container-sshnpd.yaml:8]: It's still a draft, just testing things out until I get interrupted by something

**Comment by JeremyTubongbanua** [comment]
> @JeremyTubongbanua try the posix compliant notation: `<command> > all.txt 2>&1`

Will do

**Comment by JeremyTubongbanua** [comment]
> @JeremyTubongbanua try the posix compliant notation: `<command> > all.txt 2>&1`

image

**Comment by JeremyTubongbanua** [comment]
image

the jobs now print it to stdout (so we can see what's going on) while also printing to all.txt.

this is the strategy I used:

```
...ame -s -u -v 2>&1 | tee all.txt"
```

File: tests/end2end_tests/tests/service-container-sshnpd.yaml (hunk 1 above)
Comment (JeremyTubongbanua) [reply tests/end2end_tests/tests/service-container-sshnpd.yaml:8]: I added back `-q`

File: tests/end2end_tests/entrypoints/sshnpd_entrypoint.sh [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #!/bin/bash
 echo "SSHNPD START ENTRY"
-SSHNPD_COMMAND="$HOME/.local/bin/sshnpd -a @sshnpdatsign -m @sshnpatsign -d deviceName -s -u -v 2> err.txt"
+SSHNPD_COMMAND="$HOME/.local/bin/sshnpd -a @sshnpdatsign -m @sshnpatsign -d deviceName -s -u -v 2>&1 | tee all.txt"
 echo "Running: $SSHNPD_COMMAND"
 eval "$SSHNPD_COMMAND"
+tail -f all.txt
```
Comment (xavierchanth) [inline tests/end2end_tests/entrypoints/sshnpd_entrypoint.sh]: Doesn't doing `tee all.txt` and `tail -f all.txt` duplicate the logs?
tee writes to files and stdout, which means you don't need to tail the file as well...


```
~ 
➜ echo hello | tee world
hello

~ 
➜ tail -f world                        
hello
```
In this example, hello is echoed once, printed twice...

Comment (xavierchanth) [reply tests/end2end_tests/entrypoints/sshnpd_entrypoint.sh]: You should either do `<command> > all.txt 2>&1; tail -f all.txt` or `<command> 2>&1 all.txt | tee all.txt`... unless something weird is happening in the runner?

Comment (JeremyTubongbanua) [reply tests/end2end_tests/entrypoints/sshnpd_entrypoint.sh]: The `tail -f` in the entrypoint never actually worked at all, and I forgot to remove it. 

Just removed it in this commit https://github.com/atsign-foundation/sshnoports/pull/449/commits/1bfd62c5ba7c2cbcb7696a0d6a0138c26af954a1

**Review by xavierchanth — APPROVED** [review]

File: tests/end2end_tests/entrypoints/sshnpd_entrypoint.sh (hunk 2 above)
Comment (xavierchanth) [reply tests/end2end_tests/entrypoints/sshnpd_entrypoint.sh]: Ah probably because we are in a headless pty

**Pull request merged**